### ISSUE TITLE
Add IPFS pinning

### ISSUE THREAD

**Issue by asgeir-eth** [gnosis/tabula#35]
We should add this via the implementation-agnostic API: https://ipfs.github.io/pinning-services-api-spec/.

The user should be able to:
1. Select a pinning service or provide an endpoint to a pinning service in adherence with the pinning-services-api-spec. For instance: https://api.pinata.cloud/psa.
2. The user should provide the accessToken/API key.
~~3. Everything that's added to IPFS should be pinned using the pinning service (if the pinning service is not configurated, the user should get a warning or error that makes it hard to upload content without pinning it).~~

Both the selected service and the accessToken/API key should be saved in the user's browser storage (localStorage). This is configured on the browser level and used for all the user's interactions (any publication).

**Tabula should also be usable without a pinning service:**
- If a pinning service is configurated, images and the article text content should be uploaded to IPFS, and only a hash included in the transaction content.
- If a pinning service is not configurated, image uploads should be disabled with a warning (until #47) and article text content included directly in the `article` property of the `article/create`/`article/update` transaction (like it is today).

- [x] Design
- [x] App

**Comment by cedricwaxwing** [comment]
@juliopavila I've created the design for this, and I've also included a quick screencast walking through the flow:

https://www.figma.com/file/IaimIfw2pXyR9PlQqdo6M2/Tabula?node-id=539%3A2121
https://www.loom.com/share/1d739918ee7444fda09fb3b24e28b49c

> the user should get a warning or error that makes it hard to upload content without pinning it

@asgeir-eth Just so I understand, it is still _possible_ to publish without configuring this? If so, something like this should do the trick:
https://www.figma.com/file/IaimIfw2pXyR9PlQqdo6M2/Tabula?node-id=192%3A2435

**Comment by asgeir-eth** [comment]
Nice work! 🤩

As you said, having both the pinning service name and the API endpoint seems redundant. I think it's better if we just keep the API endpoint. Or else we will have to find and list all services for no apparent reason :P 

Also, since the pinning service information is stored in localeStorage, we should not show the "set up pinning service" screen if the pinning service information is in localeStorage. I assume that was the thought, but you did not mention it explicitly in the video, so just clarifying.

> @asgeir-eth Just so I understand, it is still _possible_ to publish without configuring this? If so, something like this should do the trick: https://www.figma.com/file/IaimIfw2pXyR9PlQqdo6M2/Tabula?node-id=192%3A2435

In regards to this, I created this issue: #46 (I did not remember that we had not implemented that yet). When we have that implemented the warning should probably say something like: "you can publish, but it will be much more expensive...".

This way Tablua will be usable both with and without configuring a pinning service. **However, currently all image uploads are done via IPFS, so we should probably disable all image uploads if no pinning service is configured**, and display the warning.

**Comment by cedricwaxwing** [comment]
Great, thanks @asgeir-eth

@juliopavila I've updated the figma by removing the service dropdown and just including nickname, endpoint, and access token.

I've also created another piece of the flow, which I was thinking would be helpful. Essentially, if the user selects "Don't use IPFS", we can add a little more friction and make it clear that their transactions will be more expensive.
https://www.figma.com/file/IaimIfw2pXyR9PlQqdo6M2/Tabula?node-id=551%3A1903
